.github/workflows/docker.yml

```diff
@@ -0,0 +1,24 @@
+name: Build and publish a Docker image to ghcr.io
```

Comment:
Can we prevent this from running if the build fails?

Reply:
If you mean prevent running if the normal TypeScript action build fails, I don't think that's possible. We'd have to merge the two GitHub actions together.

Reply:
I think we can, I'll do some poking

Reply:
Looking into this more, I think we can put both of those steps in one action & skip the Docker step if the build step fails. We also won't have to build each commit twice.